core/spec/models/spree/stock/estimator_spec.rb

```diff
@@ -140,6 +140,35 @@ module Stock
             expect(shipping_rates.first.tax_rate).to eq(tax_rate)
           end
         end
+
+        it 'uses the configured shipping rate selector' do
+          DEFAULT_SHIPPING_RATE = Spree::ShippingRate.new
+
+          selector_class = Class.new do
+            def initialize(_); end;
+
+            def find_default
+              DEFAULT_SHIPPING_RATE
+            end
+          end
+          Spree::Config[:shipping_rate_selector_class] = selector_class
+
+          subject.shipping_rates(package)
+
+          expect(DEFAULT_SHIPPING_RATE.selected).to eq(true)
```

Comment:
@jordan-brough I chose to use a constant instead of lets in this case to do some state verification. Hope it's OK :)